molo/core/models.py

```diff
@@ -203,6 +345,38 @@ def get_absolute_url(self):  # pragma: no cover
     def get_parent_section(self):
         return SectionPage.objects.all().ancestor_of(self).last()
 
+    def get_effective_commenting_settings(self):
+        # return commenting settings for this section
+        if (self.commenting_state and self.commenting_state != ""):
+            return {
+                'state': self.commenting_state,
+                'open_time': self.commenting_open_time,
+                'close_time': self.commenting_close_time
+            }
+        # use the commenting settings for the parent section
+        return self.get_parent_section().get_effective_commenting_settings()
+
+    def allow_commenting(self):
+        commenting_settings = self.get_effective_commenting_settings()
+        if (commenting_settings['state'] != 'O'):  # if commenting is not open
```

Comment:
It may be worthwhile turning these `"O"` and `"T"` etc.. into constants defined on the module.